crypto/src/traits.rs

```diff
@@ -98,13 +98,17 @@ pub trait VerifyingKey:
     const LENGTH: usize;
 
     // Expected to be overridden by implementations
-    fn verify_batch(msg: &[u8], pks: &[Self], sigs: &[Self::Sig]) -> Result<(), signature::Error> {
+    fn verify_batch_empty_fail(msg: &[u8], pks: &[Self], sigs: &[Self::Sig]) -> Result<(), eyre::Report> {
```

Comment:
I'd add a comment on what is the expected behavior (ie for empty batches), so that implementations cannot miss it.